app.test.js

```diff
@@ -97,24 +100,91 @@ describe("GET /api/articles", () => {
     return request(app).get("/api/arcwtles").expect(404);
   });
 });
+
 describe("GET /api/articles/:article_id/comments", () => {
   test("should respond with an array of comments", () => {
     return request(app)
       .get("/api/articles/3/comments")
       .expect(200)
       .then((response) => {
-        expect(response.body.length).toBe(2);
+        expect(response.body.comments.length).toBe(2);
       });
   });
-  test("should respond with a 404 and a custom error message if article id requested does not exist", () => {
+  test('should respond with a 200 if an article exists but there are no comments. Returns an empty array.', () => {
+    return request(app).get("/api/articles/2/comments").expect(200).then((response) => {
+        expect(response.body.comments.length).toBe(0) 
+    })
+  });
+  test('should respond with a 400 if article_id is not a valid data type', () => {
+    return request(app).get("/api/articles/notValidDataType/comments").expect(400).then((response) => {
+        expect(response.body.msg).toBe("Invalid article");
+    })
+  });
+  test("should respond with a 400 and a custom error message if article id requested does not exist", () => {
     return request(app)
       .get("/api/articles/9999/comments")
-      .expect(404)
+      .expect(400)
       .then((response) => {
         expect(response.body.msg).toBe("Article does not exist");
       });
   });
 });
 
-/** */
+describe("POST /api/articles/:article_id/comments", () => {
+  test("should respond with successfully inserted comment on a specified article", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "test example body",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toHaveProperty("comment_id");
+        expect(response.body.comment).toHaveProperty(
+          "author",
+          newComment.username
+        );
+        expect(response.body.comment).toHaveProperty("body", newComment.body);
+      });
+  });
+  test("should respond with a 400 if either username or body are empty values", () => {
+    const newComment = {
+      username: "icellusedkars",
+      body: "",
+    };
+
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Username and body are required fields");
+      });
+  });
+  test("should respond with a 400 if username does not exist", () => {
```

Comment:
we can also have a test for if either the user or the body are not provided, as well as if the article id they are posting the comment is either invalid and if it's valid but non-existent